image_client/picturae_importer.py

```diff
@@ -24,7 +24,7 @@ class PicturaeImporter(Importer):
            along with attached images
     """
 
-    def __init__(self, paths, date_string=None):
+    def __init__(self, paths, date_string=None, istesting=False):
```

Comment:
adding this testing T/F, to prevent the init from automatically running every method, allowing me to run individual methods separately in unittests. There might be a better way of doing this?

Reply:
I am not really clear on what this would look like. Would it be like making a function in my picturae importer class that initializes all the variables I need, then inheriting picturae_importer into my test class but not super initing it, and rather running the new function in my SetUp everytime i need it?

Reply:
more or less:

from picturae_importer import PicturaeImporter

class TestPicturaeImporter(PicturaeImporter):
    def __init__(self):
        super.not_the_init_function_but_the_other_stuff()
        

class TestPicturaeImporter:
    def test_thingy(self):
        picturae_importer = TestPicturaeImporter()
        picturae_importer.run_timestamps()

and then break up picturae_importer into at least two functions. Put the stuff you don't want to hit during your testing into one of them, invoke both from the __init__() in PicturaeImporter. Put the other stuff in not_the_init_function_but_the_other_stuff.